OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -239,7 +242,7 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr, World w, ActorInfo ai,
 			if (!RequiresBaseProvider)
 				return SpriteRenderable.None;
 
-			return w.ActorsWithTrait<BaseProvider>().SelectMany(a => a.Trait.RangeCircleRenderables(wr));
+			return w.ActorsWithTrait<BaseProvider>().Where(a => !a.Trait.IsTraitDisabled).SelectMany(a => a.Trait.RangeCircleRenderables(wr));
```

Comment:
`RangeCircleRenderables` already includes a disabled check, so this can stay as it was.